google/cloud/spanner_v1/batch.py

```diff
@@ -215,6 +216,73 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.commit()
 
 
+class MutationGroups(_SessionWrapper):
+    """Accumulate mutations for transmission during :meth:`batch_write`.
+
+    :type session: :class:`~google.cloud.spanner_v1.session.Session`
+    :param session: the session used to perform the commit
+    """
+
+    committed = None
+
+    def __init__(self, session):
+        super(MutationGroups, self).__init__(session)
+        self._mutation_groups = []
+
+    def group(self):
+        """Returns a new mutation_group to which mutations can be added."""
+        mutation_group = BatchWriteRequest.MutationGroup()
+        self._mutation_groups.append(mutation_group)
```

Comment:
Instead of modifying the `_BatchBase` class can we do this?
```suggestion
        batchbase = _BatchBase(self._session)
        self._mutation_groups.append(
        BatchWriteRequest.MutationGroup(mutations=batchbase._mutations
        )
        return batchbase
```
The problem here is the `_mutations` is a protected variable. Can we see if there is any way to do this without modifying the base class? I will also try to see if there is any better way to do this.

Reply:
The suggested change doesn't work. I think this change needn't be blocking since it is an internal implementation detail and we can change it anytime later.